patreon_archiver/utils.py

```diff
@@ -120,10 +121,47 @@ def save_other(pdd: PostsData) -> SaveInfo:
     return SaveInfo(post_data_dict=pdd, target_dir=other)
 
 
+def save_podcast(session: requests.Session, pdd: PostsData) -> SaveInfo:
+    """Save podcast posts."""
+    log.debug('Podcast: %s', pdd['attributes']['url'])
+    target_dir = Path('.', 'podcasts', pdd['id'])
+    target_dir.mkdir(parents=True, exist_ok=True)
+    write_if_new(target_dir.joinpath('post.json'), f'{json.dumps(pdd, sort_keys=True, indent=2)}\n')
+
+    media_list = pdd.get('relationships', {}).get('media', {}).get('data', [])
+    for index, media_ref in enumerate(media_list, start=1):
+        media_id = media_ref['id']
+        with session.get(f'{MEDIA_URI}/{media_id}') as req:
+            media: MediaData = req.json()['data']
+            download_url = media['attributes'].get('download_url')
+            if download_url:
+                file_name = Path(media['attributes'].get('file_name') or media_id).name
+                with session.get(download_url) as req2:
+                    write_if_new(target_dir.joinpath(f'{media_id}-{file_name}'), req2.content, 'wb')
+            elif media['attributes'].get('image_urls'):
+                image_url = media['attributes']['image_urls'].get('original')
+                if image_url:
+                    ext = get_extension(media['attributes']['mimetype'])
+                    with session.get(image_url) as req2:
+                        write_if_new(
+                            target_dir.joinpath(f'{index:02d}-{media_id}.{ext}'),
+                            req2.content,
+                            'wb',
+                        )
```

Comment:
Combine the conditions:

```suggestion
            elif media['attributes'].get('image_urls') and (image_url := media['attributes']['image_urls'].get('original')):
                ext = get_extension(media['attributes']['mimetype'])
                with session.get(image_url) as req2:
                    write_if_new(
                        target_dir.joinpath(f'{index:02d}-{media_id}.{ext}'),
                        req2.content,
                        'wb',
                    )
```

This section is not covered by the tests.